src/math_util_2.c

```diff
@@ -478,117 +479,124 @@ UNUSED void func_80041A70(void) {
 
 }
 
-void func_80041A78(Mat4 arg0, s32 arg1, s32 arg2) {
+void mtfx_translation_x_y(Mat4 arg0, s32 x, s32 y) {
     arg0[0][0] = 1.0f;
     arg0[1][1] = 1.0f;
     arg0[2][2] = 1.0f;
     arg0[1][0] = 0.0f;
     arg0[2][0] = 0.0f;
     arg0[0][1] = 0.0f;
-    arg0[3][0] = arg1;
+    arg0[3][0] = x;
     arg0[2][1] = 0.0f;
     arg0[0][2] = 0.0f;
     arg0[1][2] = 0.0f;
     arg0[3][2] = 0.0f;
-    arg0[3][1] = arg2;
+    arg0[3][1] = y;
     arg0[0][3] = 0.0f;
     arg0[1][3] = 0.0f;
     arg0[2][3] = 0.0f;
     arg0[3][3] = 1.0f;
-}
 
-void func_80041AD8(Mat4 arg0, u16 arg1) {
-    f32 sp1C = sins(arg1);
-    f32 temp_f0 = coss(arg1);
-    
-    arg0[0][0] = temp_f0;
-    arg0[1][0] = -sp1C;
-    arg0[1][1] = temp_f0;
-    arg0[0][1] = sp1C;
-    arg0[2][0] = 0.0f;
-    arg0[3][0] = 0.0f;
-    arg0[2][1] = 0.0f;
-    arg0[3][1] = 0.0f;
-    arg0[0][2] = 0.0f;
-    arg0[1][2] = 0.0f;
-    arg0[3][2] = 0.0f;
-    arg0[0][3] = 0.0f;
-    arg0[1][3] = 0.0f;
-    arg0[2][3] = 0.0f;
-    arg0[2][2] = 1.0f;
-    arg0[3][3] = 1.0f;
-}
-
-void func_80041B68(Mat4 arg0, f32 arg1) {
-    arg0[1][0] = 0.0f;
-    arg0[2][0] = 0.0f;
-    arg0[3][0] = 0.0f;
-    arg0[0][1] = 0.0f;
-    arg0[2][1] = 0.0f;
-    arg0[3][1] = 0.0f;
-    arg0[0][2] = 0.0f;
-    arg0[1][2] = 0.0f;
-    arg0[3][2] = 0.0f;
-    arg0[0][3] = 0.0f;
-    arg0[1][3] = 0.0f;
-    arg0[2][3] = 0.0f;
-    arg0[2][2] = 1.0f;
-    arg0[3][3] = 1.0f;
-    arg0[0][0] = arg1;
-    arg0[1][1] = arg1;
-}
-
-void func_80041BBC(Mat4 arg0, u16 arg1, f32 arg2) {
-    f32 sp1C = sins(arg1);
-    f32 temp_f12 = coss(arg1) * arg2;
-
-    arg0[2][0] = 0.0f;
-    arg0[0][0] = temp_f12;
-    arg0[1][1] = temp_f12;
-    arg0[2][1] = 0.0f;
-    arg0[1][0] = -sp1C * arg2;
-    arg0[0][2] = 0.0f;
-    arg0[1][2] = 0.0f;
-    arg0[0][1] = sp1C * arg2;
-    arg0[3][2] = 0.0f;
-    arg0[0][3] = 0.0f;
-    arg0[1][3] = 0.0f;
-    arg0[2][3] = 0.0f;
-    arg0[3][0] = 1.0f;
-    arg0[3][1] = 1.0f;
-    arg0[2][2] = 1.0f;
-    arg0[3][3] = 1.0f;
+    /*
+     * 1 0 0 x
+     * 0 1 0 y
+     * 0 0 1 0
+     * 0 0 0 1    
+     */
+}
+
+void mtxf_u16_rotate_z(Mat4 dest, u16 angle) {
```

Comment:
I like the name. I believe angles in n64 are always u16.

Just wondering if `mtxf_rotate_z_from_angle` might be better? But your name is also good.

Reply:
I name u16 because you have a function who use instead s16 so to differenciate between this two function I add u16/s16

Reply:
good good